src/App.jsx

```diff
@@ -14,45 +16,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHA = 'alpha';
+const SORT_BY_LENGTH = 'length';
+const REVERSE = 'reverse';
+
+function gerPreparedGoods(goods, sortField, reverseField) {
+  const preraredGoods = [...goods];
+
+  if (sortField) {
+    preraredGoods
+      .sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_BY_ALPHA:
+            return good1.localeCompare(good2);
+
+          case SORT_BY_LENGTH:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      });
+  }
+
+  if (reverseField) {
+    preraredGoods.reverse();
+  }
+
+  return preraredGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+
+  const visibleGoods = gerPreparedGoods(
+    goodsFromServer, sortField, reverseField,
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_BY_ALPHA)}
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_BY_ALPHA && sortField !== REVERSE,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH && sortField !== REVERSE,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => (setReverseField(!reverseField))}
```

Comment:
It's better to use callback as argument for your setter if you need just modification based on the existed state:
```
() => setReverseField(state => !state)
```